sklearn/metrics/tests/test_score_objects.py

```diff
@@ -487,7 +488,10 @@ def check_scorer_memmap(scorer_name):
     if scorer_name in MULTILABEL_ONLY_SCORERS:
         score = scorer(estimator, X_mm, y_ml_mm)
     else:
-        score = scorer(estimator, X_mm, y_mm)
+        if scorer_name in NONZERO_Y_SCORERS:
+            score = scorer(estimator, X_mm, y_mm+1)
+        else:
+            score = scorer(estimator, X_mm, y_mm)
```

Comment:
I suspect you can use y_mm+1 in all cases

Reply:
I actually tried that, and doing so raises 5 errors 

```
======================================================================
ERROR: sklearn.metrics.tests.test_score_objects.test_scorer_memmap_input('balanced_accuracy',)
----------------------------------------------------------------------
....
ValueError: Balanced accuracy is only meaningful for binary classification problems.

======================================================================
ERROR: sklearn.metrics.tests.test_score_objects.test_scorer_memmap_input('precision',)
----------------------------------------------------------------------
...
ValueError: Target is multiclass but average='binary'. Please choose another average setting.

======================================================================
ERROR: sklearn.metrics.tests.test_score_objects.test_scorer_memmap_input('average_precision',)
----------------------------------------------------------------------
....
ValueError: Data is not binary and pos_label is not specified

======================================================================
ERROR: sklearn.metrics.tests.test_score_objects.test_scorer_memmap_input('recall',)
----------------------------------------------------------------------
...
ValueError: Target is multiclass but average='binary'. Please choose another average setting.

======================================================================
ERROR: sklearn.metrics.tests.test_score_objects.test_scorer_memmap_input('f1',)
----------------------------------------------------------------------
....
ValueError: Target is multiclass but average='binary'. Please choose another average setting.

```